Can't start the demo client

Hi,

I am following the README to start the demo.
The postgresql db and the server starts correctly with no errors.

The client fails with:

```
~/w/lyceum master NIX
- client-release
cd client && zig build run --release=fast
run
└─ run lyceum-client
   └─ zig build-exe lyceum-client ReleaseFast native
      └─ zig build-lib raylib ReleaseFast native 20 errors
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:
#include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:
#include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:10: note: in file included from /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:
#include "raygui.h"
         ^
/home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:10: note: in file included from /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:
#include <stdio.h>              // Required for: FILE, fopen(), fclose(), fprintf(), feof(), fscanf(), vsprintf() [GuiLoadStyle(), GuiLoadIcons()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:10: note: in file included from /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:
#include "raygui.h"
         ^
/home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:10: note: in file included from /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:
#include <stdio.h>              // Required for: FILE, fopen(), fclose(), fprintf(), feof(), fscanf(), vsprintf() [GuiLoadStyle(), GuiLoadIcons()]
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:
#include <stdio.h>          // Required for: vsprintf()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:
#include <stdio.h>          // Required for: vsprintf()
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:
#include <stdio.h>                  // Required for: sprintf() [Used in OpenURL()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:
#include <stdio.h>                  // Required for: sprintf() [Used in OpenURL()]
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rlgl.h:3948:70: error: cast to 'void *' from smaller integer type 'int'
    glVertexAttribPointer(index, compSize, type, normalized, stride, (void *)offset);
                                                                     ^~~~~~~~~
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:114:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:114:
#include "rlgl.h"                   // OpenGL abstraction layer to OpenGL 1.1, 3.3+ or ES2
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:
#include "external/glfw/src/init.c"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:
#include "internal.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:
#include "platform.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:11: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:
 #include "x11_platform.h"
          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:
#include <X11/Xcursor/Xcursor.h>
         ^
/usr/include/X11/Xcursor/Xcursor.h:26:10: note: in file included from /usr/include/X11/Xcursor/Xcursor.h:26:
#include <stdio.h>
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:
#include "external/glfw/src/init.c"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:
#include "internal.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:
#include "platform.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:11: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:
 #include "x11_platform.h"
          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:
#include <X11/Xcursor/Xcursor.h>
         ^
/usr/include/X11/Xcursor/Xcursor.h:26:10: note: in file included from /usr/include/X11/Xcursor/Xcursor.h:26:
#include <stdio.h>
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/posix_poll.c:27:9: error: '_GNU_SOURCE' macro redefined
#define _GNU_SOURCE
        ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:99:14: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:99:
    #include "external/glfw/src/posix_poll.c"
             ^
<command line>:5:9: note: previous definition is here
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:
#include <stdio.h>          // Required for: sprintf()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:
#include <stdio.h>          // Required for: sprintf()
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/par_shapes.h:1133:32: error: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648
    float r = (float) rand() / RAND_MAX;
                             ~~^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:111:14: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:111:
    #include "external/par_shapes.h"    // Shapes 3d parametric generation
             ^
/usr/include/stdlib.h:87:18: note: expanded from macro 'RAND_MAX'
#define RAND_MAX 2147483647
                 ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:
#include <stdio.h>              // Required for: sprintf() [Used in ExportImageAsCode()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:
#include <stdio.h>              // Required for: sprintf() [Used in ExportImageAsCode()]
         ^
/usr/include/features.h:481:9: error: '__GLIBC_MINOR__' macro redefined
#define __GLIBC_MINOR__ 36
        ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:3857:18: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:3857:
        #include <pthread.h>    /* Unfortunate #include, but needed for pthread_t, pthread_mutex_t and pthread_cond_t types. */
                 ^
/usr/include/pthread.h:21:10: note: in file included from /usr/include/pthread.h:21:
#include <features.h>
         ^
<command line>:1:9: note: previous definition is here
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:
#include <stdio.h>
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:
#include <stdio.h>
         ^
error: the following command failed with 20 compilation errors:
/nix/store/j9sp79vdhz4l6lfnvzhvcprhar803zr3-zig-0.13.0/bin/zig build-lib -I/nix/store/mnpvi1mvwyf0wsyy5wca3kfwxv2j4pd9-libglvnd-1.7.0-dev/include -L/nix/store/ymhcg6x5jrw3hx8ik1cji6awiybgp9f7-libglvnd-1.7.0/lib -lGL -I/nix/store/f3hb1fk2kaq0nx976ims5lqry03fqlby-libX11-1.8.10-dev/include -I/nix/store/qypxaa1d0a9vyxgcsqv2nkcdd0f8w4al-xorgproto-2024.1/include -L/nix/store/qrj5f1mb08k5jg7y9ikzb1njli0gvmxz-libX11-1.8.10/lib -lX11 -cflags -std=gnu99 -D_GNU_SOURCE -DGL_SILENCE_DEPRECATION=199309L -fno-sanitize=undefined -- /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rshapes.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c -cflags -std=gnu99 -D_GNU_SOURCE -DGL_SILENCE_DEPRECATION=199309L -fno-sanitize=undefined -- /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c -OReleaseFast -I /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/include -I /usr/include -I /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src -I /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src -D_GLFW_X11=1 -DPLATFORM_DESKTOP=1 -L /usr/lib -Mroot -lc --cache-dir /home/carloratm/w/lyceum/client/.zig-cache --global-cache-dir /home/carloratm/.cache/zig --name raylib -static --listen=- 
Build Summary: 2/8 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run lyceum-client transitive failure
   ├─ zig build-exe lyceum-client ReleaseFast native transitive failure
   │  └─ zig build-lib raylib ReleaseFast native 20 errors
   └─ install transitive failure
      └─ install lyceum-client transitive failure
         └─ zig build-exe lyceum-client ReleaseFast native (+2 more reused dependencies)
error: the following build command failed with exit code 1:
/home/carloratm/w/lyceum/client/.zig-cache/o/4e5ae92a2fffc6735304eb08de2876e1/build /nix/store/j9sp79vdhz4l6lfnvzhvcprhar803zr3-zig-0.13.0/bin/zig /home/carloratm/w/lyceum/client /home/carloratm/w/lyceum/client/.zig-cache /home/carloratm/.cache/zig --seed 0x21217b8d -Z15eb6fa0aab401ee run --release=fast
error: Recipe `client-release` failed on line 49 with exit code 1
```

Any idea? I am on Debian 12 (stable)

Cheers!

Same error:

```
~/w/lyceum master NIX
- just client
cd client && zig build run
run
└─ run lyceum-client
   └─ zig build-exe lyceum-client Debug native
      └─ zig build-lib raylib Debug native 20 errors
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:
#include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c:48:
#include <stdio.h>                      // Required for: FILE, fopen(), fseek(), ftell(), fread(), fwrite(), fprintf(), vprintf(), fclose()
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:10: note: in file included from /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:
#include "raygui.h"
         ^
/home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:10: note: in file included from /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:
#include <stdio.h>              // Required for: FILE, fopen(), fclose(), fprintf(), feof(), fscanf(), vsprintf() [GuiLoadStyle(), GuiLoadIcons()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:10: note: in file included from /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c:2:
#include "raygui.h"
         ^
/home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:10: note: in file included from /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src/raygui.h:1032:
#include <stdio.h>              // Required for: FILE, fopen(), fclose(), fprintf(), feof(), fscanf(), vsprintf() [GuiLoadStyle(), GuiLoadIcons()]
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:
#include <stdio.h>          // Required for: vsprintf()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c:69:
#include <stdio.h>          // Required for: vsprintf()
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:
#include <stdio.h>                  // Required for: sprintf() [Used in OpenURL()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:108:
#include <stdio.h>                  // Required for: sprintf() [Used in OpenURL()]
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rlgl.h:3948:70: error: cast to 'void *' from smaller integer type 'int'
    glVertexAttribPointer(index, compSize, type, normalized, stride, (void *)offset);
                                                                     ^~~~~~~~~
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:114:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c:114:
#include "rlgl.h"                   // OpenGL abstraction layer to OpenGL 1.1, 3.3+ or ES2
         ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:
#include "external/glfw/src/init.c"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:
#include "internal.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:
#include "platform.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:11: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:
 #include "x11_platform.h"
          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:
#include <X11/Xcursor/Xcursor.h>
         ^
/usr/include/X11/Xcursor/Xcursor.h:26:10: note: in file included from /usr/include/X11/Xcursor/Xcursor.h:26:
#include <stdio.h>
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:73:
#include "external/glfw/src/init.c"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/init.c:28:
#include "internal.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/internal.h:325:
#include "platform.h"
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:11: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/platform.h:81:
 #include "x11_platform.h"
          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/x11_platform.h:36:
#include <X11/Xcursor/Xcursor.h>
         ^
/usr/include/X11/Xcursor/Xcursor.h:26:10: note: in file included from /usr/include/X11/Xcursor/Xcursor.h:26:
#include <stdio.h>
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/src/posix_poll.c:27:9: error: '_GNU_SOURCE' macro redefined
#define _GNU_SOURCE
        ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:99:14: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c:99:
    #include "external/glfw/src/posix_poll.c"
             ^
<command line>:5:9: note: previous definition is here
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:
#include <stdio.h>          // Required for: sprintf()
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:56:
#include <stdio.h>          // Required for: sprintf()
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/par_shapes.h:1133:32: error: implicit conversion from 'int' to 'float' changes value from 2147483647 to 2147483648
    float r = (float) rand() / RAND_MAX;
                             ~~^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:111:14: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c:111:
    #include "external/par_shapes.h"    // Shapes 3d parametric generation
             ^
/usr/include/stdlib.h:87:18: note: expanded from macro 'RAND_MAX'
#define RAND_MAX 2147483647
                 ^
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:
#include <stdio.h>              // Required for: sprintf() [Used in ExportImageAsCode()]
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c:79:
#include <stdio.h>              // Required for: sprintf() [Used in ExportImageAsCode()]
         ^
/usr/include/features.h:481:9: error: '__GLIBC_MINOR__' macro redefined
#define __GLIBC_MINOR__ 36
        ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:3857:18: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:3857:
        #include <pthread.h>    /* Unfortunate #include, but needed for pthread_t, pthread_mutex_t and pthread_cond_t types. */
                 ^
/usr/include/pthread.h:21:10: note: in file included from /usr/include/pthread.h:21:
#include <features.h>
         ^
<command line>:1:9: note: previous definition is here
/usr/include/stdio.h:205:27: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
                          ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:
#include <stdio.h>
         ^
/usr/include/stdio.h:210:33: error: use of undeclared identifier 'L_tmpnam'
extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
                                ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c:180:
#include "external/miniaudio.h"         // Audio device initialization and management
         ^
/home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:10: note: in file included from /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/miniaudio.h:11447:
#include <stdio.h>
         ^
error: the following command failed with 20 compilation errors:
/nix/store/j9sp79vdhz4l6lfnvzhvcprhar803zr3-zig-0.13.0/bin/zig build-lib -I/nix/store/mnpvi1mvwyf0wsyy5wca3kfwxv2j4pd9-libglvnd-1.7.0-dev/include -L/nix/store/ymhcg6x5jrw3hx8ik1cji6awiybgp9f7-libglvnd-1.7.0/lib -lGL -I/nix/store/f3hb1fk2kaq0nx976ims5lqry03fqlby-libX11-1.8.10-dev/include -I/nix/store/qypxaa1d0a9vyxgcsqv2nkcdd0f8w4al-xorgproto-2024.1/include -L/nix/store/qrj5f1mb08k5jg7y9ikzb1njli0gvmxz-libX11-1.8.10/lib -lX11 -cflags -std=gnu99 -D_GNU_SOURCE -DGL_SILENCE_DEPRECATION=199309L -fno-sanitize=undefined -- /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rcore.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/utils.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/raudio.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rmodels.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rshapes.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtext.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rtextures.c /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/rglfw.c -cflags -std=gnu99 -D_GNU_SOURCE -DGL_SILENCE_DEPRECATION=199309L -fno-sanitize=undefined -- /home/carloratm/w/lyceum/client/.zig-cache/o/ec6be00b9a4f00e4bc20631af1f78af7/raygui.c -ODebug -I /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src/external/glfw/include -I /usr/include -I /home/carloratm/.cache/zig/p/1220aa75240ee6459499456ef520ab7e8bddffaed8a5055441da457b198fc4e92b26/src -I /home/carloratm/.cache/zig/p/122002d98ca255ec706ef8e5497b3723d6c6e163511761d116dac3aee87747d46cf1/src -D_GLFW_X11=1 -DPLATFORM_DESKTOP=1 -L /usr/lib -Mroot -lc --cache-dir /home/carloratm/w/lyceum/client/.zig-cache --global-cache-dir /home/carloratm/.cache/zig --name raylib -static --listen=- 
Build Summary: 2/8 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run lyceum-client transitive failure
   ├─ zig build-exe lyceum-client Debug native transitive failure
   │  └─ zig build-lib raylib Debug native 20 errors
   └─ install transitive failure
      └─ install lyceum-client transitive failure
         └─ zig build-exe lyceum-client Debug native (+2 more reused dependencies)
error: the following build command failed with exit code 1:
/home/carloratm/w/lyceum/client/.zig-cache/o/4e5ae92a2fffc6735304eb08de2876e1/build /nix/store/j9sp79vdhz4l6lfnvzhvcprhar803zr3-zig-0.13.0/bin/zig /home/carloratm/w/lyceum/client /home/carloratm/w/lyceum/client/.zig-cache /home/carloratm/.cache/zig --seed 0xb42a68f0 -Zaa1066aeb6528447 run
error: Recipe `client` failed on line 46 with exit code 1
```

Don't know if it helps, but if I try to build the client outside of the nix shell, I get this:

```
~/w/lyceum master
- cd client && zig build run
./build.zig:6:21: error: container 'std' has no member called 'Build'
pub fn build(b: *std.Build) !void {
                    ^
/usr/lib/zig/0.6.0/lib/zig/std/special/build_runner.zig:131:35: note: referenced here
    switch (@typeInfo(@TypeOf(root.build).ReturnType)) {
```

--- a/.github/workflows/client_build.yml
+++ b/.github/workflows/client_build.yml
@@ -42,8 +42,9 @@ jobs:
 
       - name: "[Client] Build"
         run: |
-          nix develop .#ci --impure -c just client-build-ci
+          nix build .#client
 
       - name: "[Client] Tests"
         run: |
+          nix develop .#ci --impure -c just client-build-ci
           nix develop .#ci --impure -c just client-test-ci

--- a/README.org
+++ b/README.org
@@ -65,34 +65,35 @@ To run our DEMO you need to follow the steps in this particular order:
 1. ~nix develop --impure~ (to enter a development shell)
 2. ~devenv up~ to launch a local ~postgres~
 3. ~just db-reset~ or ~db-reset~, to initialize ~postgres~ for local development
-5. Run the ~erlang~ backend in a separate shell with either:
+4. Run the ~erlang~ backend in a separate shell with either:
 
-#+begin_src shell
-   server
-   # or
-   just server
-   # or if you want a pure build
-   nix build .#server
-   ./result/bin/server foreground
-#+end_src
+    #+begin_src shell
+      server
+      # or
+      just server
+      # or if you want a pure build
+      nix build .#server
+      ./result/bin/server foreground
+    #+end_src
 
-6. Followed by running the ~zig~ client in a third shell with either:
-#+begin_src shell
-  client-release
-  # or
-  just client-release
-  # or if you prefer a pure build
-  nix build .#client
-  ./result/bin/lyceum-client
-#+end_src
+5. Followed by running the ~zig~ client in a third shell with either:
+
+    #+begin_src shell
+      client-release
+      # or
+      just client-release
+      # or if you prefer a pure build
+      nix build .#client
+      ./result/bin/lyceum-client
+    #+end_src
 
 *Addendum*: We got [[https://github.com/Dr-Nekoma/lyceum/pull/83#issuecomment-2550476288][reports]] that (sometimes) non-NixOS users will have to rely on[[https://github.com/nix-community/nixGL][ nixGL]].
 
-#+begin_src shell
-  nix run --impure github:nix-community/nixGL -- ./result/bin/lyceum-client
-#+end_src
+    #+begin_src shell
+      nix run --impure github:nix-community/nixGL -- ./result/bin/lyceum-client
+    #+end_src
 
-7. Once you get the client to launch, connect to ~localhost~ and you can try
+6. Once you get the client to launch, connect to ~localhost~ and you can try
    playing with any of the local credentials [[https://github.com/Dr-Nekoma/lyceum/blob/master/server/database/main.input.sql#L3][here]].
 
 ** Deployment

--- a/README.org
+++ b/README.org
@@ -87,7 +87,7 @@ To run our DEMO you need to follow the steps in this particular order:
       ./result/bin/lyceum-client
     #+end_src
 
-*Addendum*: We got [[https://github.com/Dr-Nekoma/lyceum/pull/83#issuecomment-2550476288][reports]] that (sometimes) non-NixOS users will have to rely on[[https://github.com/nix-community/nixGL][ nixGL]].
+    *Addendum*: We got [[https://github.com/Dr-Nekoma/lyceum/pull/83#issuecomment-2550476288][reports]] that (sometimes) non-NixOS users will have to rely on[[https://github.com/nix-community/nixGL][ nixGL]].
 
     #+begin_src shell
       nix run --impure github:nix-community/nixGL -- ./result/bin/lyceum-client

--- a/README.org
+++ b/README.org
@@ -20,22 +20,14 @@ There are several ways to run this project, but they all leverage [[https://nixo
 single [[https://devenv.sh/][devenv]] configured, that has all the tooling required to run this project,
 it also includes postgres.
 
-** Testing the Game (DEMO)
-
-1. ~nix develop --impure~
-2. ~devenv up~ to launch a local ~postgres~
-3. ~just db-reset~ or ~db-reset~, to initialize ~postgres~ for local development
-5. ~server~ (or ~./result/bin/server foreground~ if you want to run the pure nix build)
-6. followed by ~client-release~
-7. Once you get the client to launch, connect to ~localhost~ and you can try
-   playing with any of the local credentials [[https://github.com/Dr-Nekoma/lyceum/blob/master/server/database/main.input.sql#L3][here]].
-
 ** Development Shell
+
+To enter a development shell we assume you have at least [[https://nixos.org/][Nix]] installed:
 
 #+BEGIN_SRC shell
   # To enter the developmet shell
   nix develop --impure
-  # To spawn postgres
+  # To spawn postgres (for instance)
   devenv up -d
 #+END_SRC
 
@@ -53,16 +45,58 @@ to migrate the local dababase, you can use ~just db-up~ or ~db-up~.
 You can also build the server and related OCI images with Nix.
 
 #+BEGIN_SRC shell
-    nix build .#server
-#+END_SRC
-if you prefer running a container for the server:
-#+BEGIN_SRC shell
-    nix build .#buildImage
-    docker load < ./result
-    docker container run --network=host lyceum:latest
+  nix build .#server
 #+END_SRC
 
+if you prefer running a container for the server:
+
+#+BEGIN_SRC shell
+  nix build .#buildImage
+  docker load < ./result
+  docker container run --network=host lyceum:latest
+#+END_SRC
+
+or if you want to build the ~zig~ client, run ~nix build .#client~.
+
+** Testing the Game
+
+To run our DEMO you need to follow the steps in this particular order:
+
+1. ~nix develop --impure~ (to enter a development shell)
+2. ~devenv up~ to launch a local ~postgres~
+3. ~just db-reset~ or ~db-reset~, to initialize ~postgres~ for local development
+5. Run the ~erlang~ backend in a separate shell with either:
+
+#+begin_src shell
+   server
+   # or
+   just server
+   # or if you want a pure build
+   nix build .#server
+   ./result/bin/server foreground
+#+end_src
+
+6. Followed by running the ~zig~ client in a third shell with either:
+#+begin_src shell
+  client-release
+  # or
+  just client-release
+  # or if you prefer a pure build
+  nix build .#client
+  ./result/bin/lyceum-client
+#+end_src
+
+*Addendum*: We got [[https://github.com/Dr-Nekoma/lyceum/pull/83#issuecomment-2550476288][reports]] that (sometimes) non-NixOS users will have to rely on[[https://github.com/nix-community/nixGL][ nixGL]].
+
+#+begin_src shell
+  nix run --impure github:nix-community/nixGL -- ./result/bin/lyceum-client
+#+end_src
+
+7. Once you get the client to launch, connect to ~localhost~ and you can try
+   playing with any of the local credentials [[https://github.com/Dr-Nekoma/lyceum/blob/master/server/database/main.input.sql#L3][here]].
+
 ** Deployment
+
 This game is deployed in our [[https://github.com/Dr-Nekoma/trashcan][NixOS server]].
 
 * Assets

--- a/client/build.zig
+++ b/client/build.zig
@@ -52,6 +52,12 @@ pub fn build(b: *std.Build) !void {
 
     exe.step.dependOn(&assets.step);
 
+    // https://ziglang.org/learn/build-system/#conditional-compilation
+    const assets_opt = b.option([]const u8, "assets", "custom path for the assets directory") orelse "./assets";
+    const options = b.addOptions();
+    options.addOption([]const u8, "assets", assets_opt);
+    exe.root_module.addOptions("build_options", options);
+
     if (b.lazyImport(@This(), "zerl")) |zerl_build| {
         if (std.posix.getenv("LIBRARY_PATH")) |lib_path| {
             try zerl_build.add_erlang_paths(b, lib_path);

--- a/client/build.zig
+++ b/client/build.zig
@@ -44,6 +44,14 @@ pub fn build(b: *std.Build) !void {
         exe.root_module.addImport("zerl", zerl.module("zerl"));
     }
 
+    const assets = b.addInstallDirectory(.{
+        .source_dir = b.path("assets"),
+        .install_dir = .prefix,
+        .install_subdir = "assets",
+    });
+
+    exe.step.dependOn(&assets.step);
+
     if (b.lazyImport(@This(), "zerl")) |zerl_build| {
         if (std.posix.getenv("LIBRARY_PATH")) |lib_path| {
             try zerl_build.add_erlang_paths(b, lib_path);

--- a/client/src/assets.zig
+++ b/client/src/assets.zig
@@ -5,8 +5,9 @@ const rl = @import("raylib");
 const rm = rl.math;
 const std = @import("std");
 const GameState = @import("game/state.zig");
+const build_options = @import("build_options");
 
-const base_filepath = "./assets/";
+const base_filepath = build_options.assets;
 
 fn load(
     comptime T: type,

--- a/flake.nix
+++ b/flake.nix
@@ -229,6 +229,7 @@
             zigBuildFlags = [
               "-fsys=raylib"
               "--release=fast"
+              "-Dassets=${builtins.toString ./client}/assets"
             ];
 
             nativeBuildInputs = [

--- a/flake.nix
+++ b/flake.nix
@@ -237,14 +237,9 @@
             ];
 
             buildInputs =
-              with pkgs;
-              [
-                raylib
-                zigLatest
-                erlangLatest
-              ]
-              ++ lib.optionals stdenv.isLinux (linuxPkgs)
-              ++ lib.optionals stdenv.isDarwin darwinPkgs;
+              with pkgs; [ raylib zigLatest erlangLatest ]
+                ++ lib.optionals stdenv.isLinux (linuxPkgs)
+                ++ lib.optionals stdenv.isDarwin darwinPkgs;
 
             postPatch = ''
               ln -s ${pkgs.callPackage ./client/zon-deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
@@ -284,14 +279,6 @@
               rsync
               zigLatest
               raylib
-            ];
-          };
-
-          # `nix develop .#server`
-          server = pkgs.mkShell {
-            buildInputs = with pkgs; [
-              erlangLatest
-              rebar3
             ];
           };
 

